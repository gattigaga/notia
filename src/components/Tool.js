import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Icon from "components/Icon";
import Tooltip from "components/Tooltip";

const StyledIcon = styled(Icon)`
  color: #aaa;
  font-size: 24px;
  user-select: none;
`;

const StyledTooltip = styled(Tooltip)``;

const Container = styled.div`
  position: relative;
  width: 48px;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;

  &:hover ${StyledIcon} {
    color: #555;
  }

  &:hover ${StyledTooltip} {
    top: 100%;
    opacity: 1;
  }
`;

const Tool = ({ icon, tooltip, onClick }) => (
  <Container onClick={onClick}>
    <StyledIcon name={icon} />
    <StyledTooltip text={tooltip} />
  </Container>
);

Tool.propTypes = {
  icon: PropTypes.string.isRequired,
  tooltip: PropTypes.string.isRequired,
  onClick: PropTypes.func
};

export default Tool;
