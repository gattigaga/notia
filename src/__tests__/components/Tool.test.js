import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Tool from "components/Tool";

let wrapper, onClick;

beforeEach(() => {
  onClick = jest.fn();
  wrapper = shallow(<Tool icon="list" tooltip="New List" onClick={onClick} />);
});

afterEach(() => {
  onClick.mockReset();
});

describe("Tool", () => {
  it("renders correctly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("calls 'onClick'", () => {
    expect(onClick.mock.calls.length).toEqual(0);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
