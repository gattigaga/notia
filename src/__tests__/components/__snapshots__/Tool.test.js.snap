// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tool renders correctly 1`] = `
<styled.div
  onClick={[Function]}
>
  <Styled(Icon)
    name="list"
  />
  <Styled(Tooltip)
    text="New List"
  />
</styled.div>
`;
